### DISCUSSION TITLE
Option to not overwrite existing kb_id in entity_linker

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#9579] [New Features & Project Ideas]
The [EntityLinker](https://spacy.io/api/entitylinker) sets the `kb_id` for all entities, regardless of whether the entity has a `kb_id` or not. This is inconvenient in pipelines where some kb_ids may already be set by an earlier pipeline component.

I'd like an option in the `entity_linker` to not overwrite existing `kb_id`s. This could be like the `entity_recognizer`, which _never_ overwrites existing entity labels, or `entity_ruler`, which has a [config about overwriting entity labels](https://spacy.io/api/entityruler#config).

**Comment by polm** [comment]
Thanks for the suggestion. We've recently added overwrite-related settings to the Morphologizer, and as part of that discussion we realized it would be good to have that option uniformly for all components, so we'd definitely be up for adding this.

**Comment by adrianeboyd** [comment]
You're in luck: this has already been implemented in #9050, it will be in v3.2.0.

**Reply by kinghuang** [reply]
Awesome. Thanks, all! 🎉
